Sources/SwiftDiagnostics/DiagnosticsFormatter.swift

```diff
@@ -299,12 +307,14 @@ public struct DiagnosticsFormatter {
       }
 
       let columnsWithDiagnostics = Set(
-        annotatedLine.diagnostics.map {
-          annotatedLine.characterColumn(ofUtf8Column: $0.location(converter: slc).column)
+        annotatedLine.diagnostics.compactMap {
+          $0.location(converter: slc).map {
+            annotatedLine.characterColumn(ofUtf8Column: $0.column)
+          }
```

Comment:
Maybe it’s just me but I find `Optional.map` always hard to read and would prefer.

```suggestion
          if guard let location = $0.location(converter: slc) else {
            return nil
          }
          return annotatedLine.characterColumn(ofUtf8Column: $0.column)
```